lib/ansible/modules/system/hostname.py

```diff
@@ -579,7 +579,13 @@ class SLESHostname(Hostname):
 
 class OpenSUSEHostname(Hostname):
     platform = 'Linux'
-    distribution = 'Opensuse '
+    distribution = 'openSUSE Leap'
+    strategy_class = SystemdStrategy
+
+
+class TumbleweedHostname(Hostname):
+    platform = 'Linux'
+    distribution = 'openSUSE Tumbleweed'
```

Comment:
I have to make this change to make it work on OpenSUSE Tumbleweed - Could you please confirm ? 
```suggestion
    distribution = 'Opensuse tumbleweed'
```

Reply:
Do you have `distro` package installed ?

Reply:
Currently this value will be `Opensuse tumbleweed` as @Akasurde describes since we are using `distro.name()` in `lib/ansible/module_utils/common/sys_info.py`. I have a PR open (#52199) switching to `distro.id()` which provides more reliable and consistent output for identifying a distribution. That would make the distribution `Opensuse-tumbleweed`.

OpenSUSE seems to be the first distribution I have encountered that returns a two part id. I'm wondering if it would be nicer to camel case the distribution so we would get `OpensuseTumbleweed`.

We should hold off merging this PR until #52199 gets merged. Then the distribution string will be correct and we can match on that in this PR.

Reply:
@Akasurde sorry, yes, I will send a new version setting SLES and openSUSE variants to capital version, since capitalize is used in lib/ansible/module_utils/common/sys_info.py
So I think your change for opensuse tumbleweed is also right, and so is SLES.

I'll rework my commit message, since now distro package is being used, mentioning just the distro names being changed.

Thanks for review!